### DISCUSSION TITLE
[Question] Global middleware for Queue Jobs

### DISCUSSION THREAD

**Discussion by ankurk91** [laravel/framework#46163] [Q&A] [answered]
Hi,

I am looking for way to specify the middleware for all jobs.

Right now we can add middleware to a job like:

```php
// app/Jobs/TestJob.php

public function middleware()
{
       return [new PreventDuplicateJobMiddleware];
}
```

What if we could do it globally for all Jobs/Mailables/Notifications etc

```php
// AppServiceProvider
Queue::middleware($arrayOfMiddleware)
```

How to make it possible ?

**Comment by rojtjo** [comment, marked as answer]
I'm not sure if this is documented anywhere, but you can use the `Bus::pipeThrough()` method for this:

```php
use Illuminate\Support\Facades\Bus;

Bus::pipeThrough([
    SomeMiddleware::class,
    new SomeOtherMiddlware(),
    fn ($job, $next) => $next($job),
]);
```

See:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Support/Facades/Bus.php#L21
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Bus/Dispatcher.php#L270-L281

**Reply by ankurk91** [reply]
Thanks, i will try this out and will post back here

**Reply by ankurk91** [reply]
It worked like a charm. Thank you so much.
